cmd/kube-proxy/app/server_others_test.go

```diff
@@ -450,6 +460,42 @@ func Test_getLocalDetector(t *testing.T) {
 			expected:    proxyutiliptables.NewNoOpLocalDetector(),
 			errExpected: false,
 		},
+		// LocalModeInterfaceName, nodeInfo and ipt are not needed for these cases
+		{
+			mode:        proxyconfigapi.LocalModeInterfaceName,
+			config:      &proxyconfigapi.KubeProxyConfiguration{InterfaceName: "eth"},
+			expected:    resolveLocalDetector(t)(proxyutiliptables.NewDetectLocalByInterfaceName("eth")),
+			errExpected: false,
+		},
+		{
+			mode:        proxyconfigapi.LocalModeInterfaceName,
+			config:      &proxyconfigapi.KubeProxyConfiguration{InterfaceName: ""},
+			errExpected: true,
+		},
+		{
+			mode:        proxyconfigapi.LocalModeInterfaceName,
+			config:      &proxyconfigapi.KubeProxyConfiguration{InterfaceName: "1234567890123456789"},
+			expected:    resolveLocalDetector(t)(proxyutiliptables.NewDetectLocalByInterfaceName("1234567890123456789")),
+			errExpected: false,
+		},
+		// LocalModeInterfacePrefix, nodeInfo and ipt are not needed for these cases
+		{
+			mode:        proxyconfigapi.LocalModeInterfacePrefix,
+			config:      &proxyconfigapi.KubeProxyConfiguration{InterfacePrefix: "eth"},
+			expected:    resolveLocalDetector(t)(proxyutiliptables.NewDetectLocalByInterfacePrefix("eth")),
+			errExpected: false,
+		},
+		{
+			mode:        proxyconfigapi.LocalModeInterfacePrefix,
+			config:      &proxyconfigapi.KubeProxyConfiguration{InterfacePrefix: ""},
+			errExpected: true,
+		},
+		{
+			mode:        proxyconfigapi.LocalModeInterfacePrefix,
+			config:      &proxyconfigapi.KubeProxyConfiguration{InterfacePrefix: "1234567890123456789"}, // doesn't fail because validation is done before calling getLocalDetector
```

Comment:
this is no longer true, right? there is no check for max length on validation, it only checks is not empty

Reply:
yeah good catch, I missed this in all the iterations, let me remove the comment, though I'll leave the test case there as a reminder that we don't validate...

Reply:
done/